packages/repository/src/transaction.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+/**
+ * Local transaction
+ */
+export interface Transaction {
+  /**
+   * Commit the transaction
+   */
+  commit(): Promise<void>;
+
+  /**
+   * Rollback the transaction
+   */
+  rollback(): Promise<void>;
+
+  /**
+   * The transaction Identifier
+   */
+  id: string;
+
+  /**
+   * The connection used by the transaction
+   */
+  connection: any;
```

Comment:
Let's use `unknown` instead of any. That way consumers of this property must explicitly cast it to a type they are assuming to see there, instead of being able to treat it as literally anything and thus bypassing compiler checks.

```suggestion
  connection: unknown;
```

Personally, I'd prefer to keep `connection` as an internal implementation detail and DO NOT expose it in this interface. Instead, expose `isActive` property as discussed earlier.

AFAICT, your current test suite does not need the check whether the connection is active, thus I think it will be best to leave both `connection` and `isActive` properties out of scope of this pull request.

Reply:
Yea I agree that we can revisit this in another PR (I like the idea in https://github.com/strongloop/loopback-next/pull/3397#discussion_r308605303). But the `afterEach()` hook inside `transactions.suite.ts` would need this to function properly (to check active transactions) at the moment.

Reply:
> But the `afterEach()` hook inside `transactions.suite.ts` would need this to function properly (to check active transactions) at the moment.

Ah, I see.

As a temporary solution, can we perhaps disable type checks in that hook? Something along the following lines:

```ts
afterEach(async () => {
  // FIXME: replace tx.connection with tx.isActive when it become available
  if (tx !== undefined && (tx as any).connection) {
    await tx.rollback();
  }
});
```

Thoughts?

Reply:
Nice! I'll try that :D.

Reply:
I've created https://github.com/strongloop/loopback-next/issues/3471 to track this issue.